obspy/io/sac/sactrace.py

```diff
@@ -1158,10 +1176,23 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
                                             checksize=checksize)
         if not debug_strings:
             for i, val in enumerate(hs):
-                val = _ut._clean_str(val, strip_whitespace=False)
+                if encoding.lower() != 'ascii':  # throw warning
+                    msg = ("Using encoding other than ASCII: {}\n"
+                           "All non-ASCII characters will be replaced by "
+                           "?".format(encoding))
```

Comment:
Is this error message correct? Should it not rather be `"Using non-ASCII encoding: XXX. All characters that cannot be decoded with it will be replaced by '?'"`? Also maybe warn only when this actually happens? e.g. when it fails to decode some string.